src/Microsoft.Health.Dicom.Core/Features/Store/StoreValidatorResultBuilder.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.ObjectModel;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class StoreValidatorResultBuilder
+{
+    public StoreValidatorResultBuilder()
+    {
+        ErrorMessages = new Collection<string>();
+        WarningMessages = new Collection<string>();
+        WarningCodes = ValidationWarnings.None;
+        FirstException = null;
+    }
+
+    private Collection<string> ErrorMessages { get; }
```

Comment:
Please do not use `private` properties. Instead use fields.

Reply:
Done.